homework_7/functional.py

```diff
@@ -0,0 +1,52 @@
+def func_chain(*args):
+
+    #sequential execution of all functions
+    def func(x):
+        for f in args:
+            x = f(x)
+        return x
+    
+    return func
+
+
+def sequential_map(*args):
+
+    return list(map(func_chain(*args[:-1]), args[-1]))
+
+
+def consensus_filter(*args):
+    
+    lst = args[-1]
+    
+    #filtering
+    for f in args[:-1]:
+        lst_bool = list(map(f, lst))
+        
+        for i in range(len(lst_bool)-1, -1, -1):
+            if not lst_bool[i]:
+                del lst[i]
+
+    return lst
```

Comment:
Велосипед, всё куда проще
```suggestion
    *filter_functions, container = args
    for func in filter_functions:
        container = filter(func, container)
    return list(container)
```